### PR TITLE
#minor feat: pagination support for get commands

### PR DESCRIPTION
**Pull request by nishantwrp**
Signed-off-by: Nishant Mittal <mittalnishant14@outlook.com>

# TL;DR
Add pagination support for get commands. Example

```bash
flytectl get workflows -p flytesnacks -d development  --filter.page 2
```

## Type
- [ ] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [ ] Smoke tested
- [x] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
Used the `Token` field defined in `FlyteIDL`. By testing, came to know it's actually the offset of many items to skip.

## Tracking Issue
Fixes https://github.com/flyteorg/flyte/issues/1146.

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: pkg/filters/util.go [hunk 1]
```diff
@@ -52,3 +56,8 @@ func buildSortingRequest(c Filters) *admin.Sort {
 	}
 	return nil
 }
+
+func getToken(c Filters) string {
+	token := int(c.Page-1) * int(c.Limit)
```
Comment (evalsocket) [inline pkg/filters/util.go:61]: Why do we need this ?

**Comment by evalsocket** [comment]
Thanks @nishantwrp for doing it, Please fix the unit test and documentation. Please add testing details in PR description.

File: pkg/filters/util.go (hunk 1 above)
Comment (nishantwrp) [reply pkg/filters/util.go:61]: I think the `token` is actually the offset i.e. the number of items to skip while fetching. In the issue, we wanted to build a page-based flag. So, this actually calculates the number of items to skip using `page` and `limit`.

**Comment by nishantwrp** [comment]
To manually test this PR. I first added 250 versions of a workflow and tried playing around with the page number and limit.

File: pkg/filters/util.go (hunk 1 above)
Comment (nishantwrp) [reply pkg/filters/util.go:61]: Just verified that it's the offset. https://github.com/flyteorg/flyteadmin/blob/60110119d8919403c43f74c781d59a380f3d1f21/pkg/manager/impl/task_manager.go#L166

Comment (evalsocket) [reply pkg/filters/util.go:61]: correct

**Review by evalsocket — APPROVED** [review]
Nice

**Comment by evalsocket** [comment]
@nishantwrp let's wait for @pmahindrakar-oss review, in mean time it would be great if you can also add example for new flag `page` in get command(`project/task/workflow/execution/launch_plan`) https://github.com/flyteorg/flytectl/blob/master/cmd/get/project.go#L39

**Comment by nishantwrp** [comment]
@evalsocket done.

**Review by evalsocket — APPROVED** [review]

**Pull request merged**